### PR TITLE
update references to requestSave to add .catch #9335

### LINKED ISSUE #9335: Add error conditions to all the places where we call requestSave and don’t have .catch statements
Eaxmple: https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/3617/events/latest/

AC:
Add error conditions to all the places where we call requestSave and don’t have .catch statements

Reference: 
https://dsva.slack.com/archives/C6E41RE92/p1549910375017400

### PR DESCRIPTION
**Pull request by youngfreezyVA**
Resolves #9335 

### Description
added `.catch` blocks to everywhere that `requestSave` was called. removed an unused reference in `CaseTitleDetails.jsx`

### Acceptance Criteria
- [ ] 400 errors no longer are alerted to sentry

### Testing Plan
try to enter an invalid document id in the submit view and you should see the error message appear, but the error will be console.log'ed now instead of thrown.

### CODE REVIEW

File: client/app/queue/CreateMailTaskDialog.jsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (aroltsch) [inline client/app/queue/CreateMailTaskDialog.jsx]: Could we remove this extra line.

File: client/app/queue/EvaluateDecisionView.jsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (aroltsch) [inline client/app/queue/EvaluateDecisionView.jsx]: Do you need it here?

File: client/app/queue/SelectSpecialIssuesView.jsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable no-console */
+
```
Comment (aroltsch) [inline client/app/queue/SelectSpecialIssuesView.jsx]: Could we remove this empty line.

File: client/app/queue/EvaluateDecisionView.jsx [hunk 4]
```diff
@@ -159,6 +160,9 @@ class EvaluateDecisionView extends React.PureComponent {
       then(() => this.props.deleteAppeal(appealId), (response) => {
         // eslint-disable-next-line no-console
         console.log(response);
+      }).
+      catch(() => {
+        // handle the error from the frontend
```
Comment (aroltsch) [inline client/app/queue/EvaluateDecisionView.jsx]: I think this comment is unnecessary.

Comment (youngfreezyVA) [reply client/app/queue/EvaluateDecisionView.jsx]: it's necessary for es lint, it complains about empty functions, so it's either a comment or disabling another eslint rule

File: client/app/queue/EvaluateDecisionView.jsx (hunk 2 above)
Comment (youngfreezyVA) [reply client/app/queue/EvaluateDecisionView.jsx]: yes, console.logging was already used in this component from previous changes, on line 162

File: client/app/queue/SelectSpecialIssuesView.jsx (hunk 3 above)
Comment (youngfreezyVA) [reply client/app/queue/SelectSpecialIssuesView.jsx]: done

**Pull request merged**